executor/executor.go

```diff
@@ -76,7 +76,7 @@ var (
 
 type baseExecutor struct {
 	ctx           sessionctx.Context
-	id            string
+	id            func() string
```

Comment:
Maybe define id as a type which implement Stringer method is a better way?

Reply:
+1, @lysu PTAL